src/ayab/encoders.cpp

```diff
@@ -155,6 +150,38 @@ void Encoders::encA_rising() {
       (hallValue > FILTER_L_MAX[static_cast<uint8_t>(m_machineType)])) {
     m_hallActive = Direction_t::Left;
 
+    // Only set the belt shift the first time a magnet passes the turn mark.
+    // Headed to the right.
+    if (!m_passedLeft && Direction_t::Right == m_direction) {
+      // Belt shift signal only decided in front of hall sensor
+      m_beltShift = digitalRead(ENC_PIN_C) != 0 ? BeltShift::Shifted : BeltShift::Regular;
+      m_passedLeft = true;
+
+      if (Carriage_t::Garter == m_carriage) {
+        // This has to be the first magnet and the belt shift needs to be swapped
+        // But only for the G-carraige
+        if (m_position < 30) {
+          if (BeltShift::Regular == m_beltShift) {
+            m_beltShift = BeltShift::Shifted;
+          } else {
+            m_beltShift = BeltShift::Regular;
+          }
+        }
+      }
+    }
+
+    // The garter carriage has a second set of magnets that are going to
+    // pass the sensor and will reset state incorrectly if allowed to
+    // continue.
+    if (m_carriage == Carriage_t::Garter) {
```

Comment:
Nit- in some of your comparisons you put the constant first and in some you use the variable first. I think we should prefer constant first but either way, surely we want to be consistent?

Reply:
+1 for consistency.

Although if we have to settle on one style, I'd rather have the constant on the right as above. The safety advantages of the reversed order are inexistent, since if `if (m_carriage = Carriage::Garter)` is written, the compiler will both raise a warning (which is treated as an error) and a type error because enum classes can't be implicitly converted to booleans.

As a side note, it seems to me the `Carriage_t` typedef (and others like it) only still exists for compatibility with existing code. `Carriage::Garter` works just as fine and seems cleaner; most code still uses the `_t` variants but there are many occurrences of `BeltShift::` for example — here too, consistency would be nice.